src/client/include/com/amazonaws/kinesis/video/client/Include.h

```diff
@@ -1140,6 +1142,63 @@ struct __StreamMetrics {
 
     // Last measured transfer rate in bytes per second
     UINT64 currentTransferRate;
+
+    // V1 metrics following
+
+    // Total stream duration - stream uptime
+    UINT64 uptime;
+
+    // Total transferred bytes
+    UINT64 transferredBytes;
+
+    // Total number of streaming sessions including new/errored/active
+    UINT64 totalSessions;
+
+    // Total number of active streaming sessions - only the ones that actually have streamed some data
+    UINT64 totalActiveSessions;
+
+    // Average session duration
+    UINT64 avgSessionDuration;
+
+    // Total number of buffered ACKs
+    UINT64 bufferedAcks;
+
+    // Total number of received ACKs
+    UINT64 receivedAcks;
+
+    // Total number of persisted ACKs
+    UINT64 persistedAcks;
+
+    // Total number of error ACKs
+    UINT64 errorAcks;
+
+    // Total number of dropped frames
+    UINT64 droppedFrames;
+
+    // Total number of dropped fragments
+    // TODO: Near-realtime streaming mode is not used currently supported and this value will be 0
+    UINT64 droppedFragments;
+
+    // Total number of skipped frames
+    UINT64 skippedFrames;
+
+    // Total number of latency pressure events
+    UINT64 latencyPressures;
+
+    // Total number of Buffer pressure events
+    UINT64 bufferPressures;
+
+    // Total number of stream stale events
+    UINT64 staleEvents;
+
+    // Total number of put frame call errors
+    UINT64 putFrameErrors;
+
+    // Backend Control Plane API call latency which includes success and failure
+    UINT64 cplApiCallLatency;
+
+    // Backend Data Plane API call latency which includes success and failure
+    UINT64 dataApiCallLatency;
 };

```

Comment:
Just throwing it out here. What do you think about having metrics tracking different retries? I am unsure of the different retries within the stack though.

Reply:
At this locality, we can't track the retriable/non-retriable errors for example as those depend on a number of factors including the scenario.

We should have separate metrics tracking fatal errors perhaps but those should be tracked at a higher layer.

For example. A stream could be configured to not retry. In this case a retriable error will be fatal. Another example is an auth integration - this might be retriable or fatal - depending on the integration. Similar to this is the network errors or media errors.

I believe that we can't fix every possible situation. These are the basic metrics needed. We will have to add more heuristics/higher-logic dependent metrics as an iterative process.

Reply:
Understood. Makes sense.